spec/unit/puppet_pal_2pec.rb

```diff
@@ -112,6 +112,18 @@
             expect { ctx.with_script_compiler {|c| c.evaluate_string('$a') }}.to raise_error(/Unknown variable: 'a'/)
           end
         end
+
+        it 'instantiates definitions in the given code string' do
+          result = Puppet::Pal.in_tmp_environment('pal_env', modulepath: modulepath, facts: node_facts) do |pal|
+            pal.with_script_compiler do |compiler|
+              compiler.evaluate_string(<<-CODE)
```

Comment:
Just a remark. Although using CODE here is totally fine, using PUPPET would be even better. Why? Because RubyMine actually attempts to select a  syntax checker based on that word. Same thing when the code is RUBY, YAML, JSON, or other syntax that is well known to the IDE.